infra/conf/xray.go

```diff
@@ -504,21 +504,24 @@ func (c *Config) Override(o *Config, fn string) {
 
 	// update the Outbound in slice if the only one in overide config has same tag
 	if len(o.OutboundConfigs) > 0 {
-		if len(c.OutboundConfigs) > 0 && len(o.OutboundConfigs) == 1 {
-			if idx := c.findOutboundTag(o.OutboundConfigs[0].Tag); idx > -1 {
-				c.OutboundConfigs[idx] = o.OutboundConfigs[0]
-				ctllog.Println("[", fn, "] updated outbound with tag: ", o.OutboundConfigs[0].Tag)
+		outboundPrepends := o.OutboundConfigs
```

Comment:
as you know when file name includes "tail", outbound configs will append to original config and if not, the override outbounds should prepend to original config.
in several test cases some outbounds of the original config will update if they have the same Tag name in :
```
c.OutboundConfigs[idx] = o.OutboundConfigs[i]
```
so if i use c.OutboundConfigs for prepend instead of `outboundPrepends` , it may cause of duplicate outbound with same tag name in final config.

Reply:
o.OutboundConfigs are incoming data, c.OutboundConfigs are existing data. While we iterate through o.OutboundConfigs there are three case:
1. tag match, replace one of existing
2. append
3. prepend
1 and 3 should not both happen for the same entry. Why it create duplicates?

Reply:
@yuhan6665  actually after write a lot of explanations ,suddenly realized what happened here. i think now `outboundPrepends` seems reasonable  now :)